### DISCUSSION TITLE
Are nested elements allowed in pinia

### DISCUSSION THREAD

**Discussion by david8z** [vuejs/pinia#1099] [Help and Questions] [answered]
Hi I am not sure if this is a bug or if it is intentionally working like that.

Let say I have the following object:
```javascript
items: [
  {
    colors: [
        { value: true}
     ]
  },
  {
    colors: [
        {value: true}
     ]
  }
]
```

When i change lets say `items[0].colors[0].value = false` automatically `items[1].colors[0].value` is also changed to `false`, is this an expected behaviour? 

https://user-images.githubusercontent.com/31882023/155902945-0274e9a7-af2a-40ea-aefb-9627fd33edd8.mp4

They look kind of linked as if I delte one of the with the devtools both are deleted

**Comment by soc221b** [comment, marked as answer]
It seems due to your `items[1].colors[0]` and `items[2].colors[0]` share the same object.

Maybe you forgot to **deeply** clone it?

For example, assuming you have an `addItem` function like this:

```
const initialColors = [{ active: true, hex: "000000", id: 0 }]
function addItem () {
  const newItem = {
    colors: [...initialColors], // forgot to deeply clone here
    colors: JSON.parse(JSON.stringify(initialColors)), // ok

    // ...
  }
  state.items.push(newItem)
}
```

---

Further reading: [Immerjs](https://immerjs.github.io/immer/#a-quick-example-for-comparison) is a library to help us solve this kind of problem in a elegant way.

**Reply by david8z** [reply]
Oh that is totally it many thanks :+1:

**Reply by sbay-ppm** [reply]
Forgetting to deeply clone was my issue as well.  Thank you both!
